detectable/src/main/java/com/synopsys/integration/detectable/detectables/bitbake/parse/GraphNodeLabelParser.java

```diff
@@ -15,30 +18,39 @@
 //  virtual:native:/workdir/poky/meta/recipes-support/attr/acl_2.3.1.bb -> GraphNodeLabelDetails.recipeSpec
 public class GraphNodeLabelParser {
     private static final String LABEL_PATH_SEPARATOR = "/";
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
 
-    public String parseVersionFromLabel(String label) throws IntegrationException {
-        GraphNodeLabelDetails labelDetails = parseLabelParts(label);
-        return labelDetails.getVersion();
+    public Optional<String> parseVersionFromLabel(String label) {
+        Optional<GraphNodeLabelDetails> labelDetails = parseLabelParts(label);
+        if (labelDetails.isPresent()) {
+            return Optional.of(labelDetails.get().getVersion());
+        }
+        return Optional.empty();
     }
 
-    public String parseLayerFromLabel(String label, Set<String> knownLayerNames) throws IntegrationException {
-        GraphNodeLabelDetails labelDetails = parseLabelParts(label);
-        String recipeSpec = labelDetails.getRecipeSpec();
+    public Optional<String> parseLayerFromLabel(String label, Set<String> knownLayerNames) {
+        Optional<GraphNodeLabelDetails> labelDetails = parseLabelParts(label);
+        if (!labelDetails.isPresent()) {
+            return Optional.empty();
+        }
+        String recipeSpec = labelDetails.get().getRecipeSpec();
         for (String candidateLayerName : knownLayerNames) {
             String possibleLayerPathSubstring = LABEL_PATH_SEPARATOR + candidateLayerName + LABEL_PATH_SEPARATOR;
             if (recipeSpec.contains(possibleLayerPathSubstring)) {
```

Comment:
Not a huge change, but utilizes the `.map` on Optional which I think looks a little cleaner.
```suggestion
        Optional<String> recipeSpec = parseLabelParts(label)
            .map(GraphNodeLabelDetails::getRecipeSpec);
        if (!recipeSpec.isPresent()) {
            return Optional.empty();
        }
        for (String candidateLayerName : knownLayerNames) {
            String possibleLayerPathSubstring = LABEL_PATH_SEPARATOR + candidateLayerName + LABEL_PATH_SEPARATOR;
            if (recipeSpec.get().contains(possibleLayerPathSubstring)) {
```

Reply:
```
        Optional<String> recipeSpec = parseLabelParts(label)
            .map(GraphNodeLabelDetails::getRecipeSpec);
        if (recipeSpec.isPresent()) {
            for (String candidateLayerName : knownLayerNames) {
                String possibleLayerPathSubstring = LABEL_PATH_SEPARATOR + candidateLayerName + LABEL_PATH_SEPARATOR;
                if (recipeSpec.get().contains(possibleLayerPathSubstring)) {
                    return Optional.of(candidateLayerName);
                }
            }
        }
        logger.warn("Graph Node recipe '{}' does not correspond to any known layer ({})", label, knownLayerNames);
        return Optional.empty();
```
Does this make since? You would always get the warning when a recipeSpec isn't found

Reply:
It's my paranoia over not knowing for sure what bitbake will throw at us. I don't think we should ever see that warning. By this point we know we've found a label, and I believe the value will always be the path to the recipe which should always include the layer as a dir node. If we ever see that warning, it means I've misunderstood something.